crates/detect-targets/src/lib.rs

```diff
@@ -0,0 +1,16 @@
+//! Detect the target at the runtime.
+//!
+//! Example use cases:
+//!  - The binary is built with musl libc to run on anywhere, built
+//!    the runtime supports glibc.
+//!  - The binary is built for x86_64-apple-darwin, but run on
+//!    aarch64-apple-darwin.
```

Comment:
would be good to:
- add at least one example (doesn't need to do an assert_eq like the others, just a quick "copy-paste to get started" thing)
- explain the two different APIs in here, the one that just gets the runtime targets, and the desired targets variant; and have a recommendation for what to use when

Reply:
Done!

I also realized that the API `get_desired_output` takes `&Option<String>`, which is not very ergonomic and I changed it to `Option<&str>`.